plugins/woocommerce/includes/wc-template-functions.php

```diff
@@ -2203,7 +2205,8 @@ function woocommerce_widget_shopping_cart_button_view_cart() {
 	 * Output the proceed to checkout button.
 	 */
 	function woocommerce_widget_shopping_cart_proceed_to_checkout() {
-		echo '<a href="' . esc_url( wc_get_checkout_url() ) . '" class="button checkout wc-forward">' . esc_html__( 'Checkout', 'woocommerce' ) . '</a>';
+		$wp_button_class = wc_wp_theme_get_element_class_name( 'button' ) ? ' ' . wc_wp_theme_get_element_class_name( 'button' ) : '';
+		echo '<a href="' . esc_url( wc_get_checkout_url() ) . '" class="button checkout wc-forward' . esc_attr( $wp_button_class ) . '">' . esc_html__( 'Checkout', 'woocommerce' ) . '</a>';
```

Comment:
I see this pattern repeated in several places, wondering if we could use `trim()` to simplify the code. Something along these lines:

```suggestion
		echo '<a href="' . esc_url( wc_get_checkout_url() ) . '" class="button checkout wc-forward' . esc_attr( trim( ' ' . wc_wp_theme_get_element_class_name( 'button' ) ) ) . '">' . esc_html__( 'Checkout', 'woocommerce' ) . '</a>';
```

Not sure if the code is much clearer, thought. :sweat_smile: So just leaving the suggestion here, but I don't have a strong preference anyway.

Reply:
Thanks! I was also slightly bothered by it, so was thinking about whether I should make a function out of it. The usage of `trim` is a nice trick.